driver/src/price_finding/price_finder_interface.rs

```diff
@@ -157,12 +156,41 @@ pub fn execute_private_solver(
     Ok(private_solver_command.output()?)
 }
 
-#[cfg_attr(test, automock)]
 pub trait PriceFinding {
-    fn find_prices(
-        &self,
-        orders: &[models::Order],
-        state: &models::AccountState,
+    fn find_prices<'a>(
+        &'a self,
+        orders: &'a [models::Order],
+        state: &'a models::AccountState,
         time_limit: Duration,
-    ) -> Result<models::Solution, Error>;
+    ) -> BoxFuture<'a, Result<models::Solution, Error>>;
 }
+
+// We would like to tag `PriceFinding` with `mockall::automock` but mockall does not support the
+// lifetime bounds on `tokens`: https://github.com/asomers/mockall/issues/134 . As a workaround
+// we create a similar trait with simpler lifetimes on which mockall works.
+#[cfg(test)]
+mod mock {
```

Comment:
This is the same workaround as before. I think it is the last instance where we need it.